rust/src/main.rs

```diff
@@ -57,9 +57,10 @@ fn main() -> anyhow::Result<()> {
         .merge(Toml::file(
             args.configuration_file
                 .as_deref()
-                .unwrap_or_else(|| Path::new("pjdfstest.toml")),
```

Comment:
I suggest
```rust
let config: Config = if let Some(path) = args.configuration_file.as_deref() {
    Figment::new().merge(Toml::file(path))
} else {
    Figment::new()
};
```
That way you won't ever have to construct an empty file.  You might not need to derive `Default` for all those config structs, too.

Reply:
Thanks, I applied your suggestion.